pkg/ebpf/c/tracer-ebpf.c

```diff
@@ -613,6 +665,28 @@ static int handle_retransmit(struct sock* sk, tracer_status_t* status) {
     return 0;
 }
 
+__attribute__((always_inline))
+static int handle_tcp_close(struct pt_regs* ctx,
+    struct sock* sk,
+    tracer_status_t* status) {
+    u64 pid_tgid = bpf_get_current_pid_tgid();
+
+    u32 net_ns_inum;
+
+    // Get network namespace id
+    possible_net_t* skc_net;
+
+    skc_net = NULL;
+    net_ns_inum = 0;
+    bpf_probe_read(&skc_net, sizeof(possible_net_t*), ((char*)sk) + status->offset_netns);
+    bpf_probe_read(&net_ns_inum, sizeof(net_ns_inum), ((char*)skc_net) + status->offset_ino);
+
+    log_debug("kprobe/tcp_close: pid_tgid: %d, ns: %d\n", pid_tgid, net_ns_inum);
+
+    handle_family(sk, status, cleanup_tcp_conn(ctx, sk, status, pid_tgid, family));
```

Comment:
Won't this delete the ID for the tcp_connection being closed (meaning the next close event will have another ID), or is the idea to protect from a single event sent on the perf buffer being processed multiple times?

Reply:
Yeah the goal of the id is to avoid processing multiple times the ~same close event 👍